src/docs/arch-checkpoint.dox

```diff
@@ -0,0 +1,62 @@
+/*! @arch_page arch-checkpoint Checkpoint
+
+# Overview #
+
+A checkpoint is a known point in time from which WiredTiger can recover in the event of a
+crash or unexpected shutdown. WiredTiger checkpoints are created either via the API
+WT_SESSION::checkpoint, or internally. Internally checkpoints are created on startup, shutdown
+and during compaction.
+
+A checkpoint is performed within the context of snapshot isolation transaction as such the
+checkpoint has a consistent view of the database from beginning to end. The checkpoint can be run
+with or without a timestamp, generally a timestamp is used. This timestamp is not used as a
+\c read_timestamp for the checkpoint transaction. The timestamp provided is set to the checkpoint
+timestamp which is written out with the metadata information for the checkpoint.
+
+# The checkpoint algorithm #
+
+A checkpoint can be broken up into 5 main stages:
+
+1. The prepare stage
+Checkpoint prepare sets up the checkpoint, it begins the checkpoint transaction, updates the global
+checkpoint state and gathers a list of handles to be checkpointed. Where each handle refers to a
+specific b-tree. The set of b-trees gathered by the checkpoint varies based off configuration.
+Additionally clean b-trees, i.e. b-trees without any modifications are excluded from the list.
```

Comment:
Still, clean btrees will be checkpointed under some special conditions, no need list all conditions but worth mentioning.